feat(AdvancedCalendar): update presets style, storybook args, navigation style, localization

AbdulrahmanNahhas opened the pull request:
## Description

Enhanced the layout and functionality of the Date Caption component, improving the user experience and visual appeal.

---

## Screenshots

Comparison:
Left: Before | Right: After

### Storybook Args:
<img width="1440" alt="Screenshot 2024-11-02 at 6 23 37 PM" src="https://github.com/user-attachments/assets/785a51b4-e482-4656-bea8-b9ee4f0fe68b">

### Date Caption Layouts:

#### 1. Buttons:
<img width="378" alt="Screenshot 2024-11-02 at 6 26 42 PM" src="https://github.com/user-attachments/assets/6d9e573f-c40f-4e62-8c34-02a1db77b419">

#### 2. Dropdown:
<img width="378" alt="Screenshot 2024-11-02 at 6 27 00 PM" src="https://github.com/user-attachments/assets/3bcf47f3-88d9-4a7a-b457-a2d9c3d13af0">

#### 3. Dropdown and Buttons:
<img width="378" alt="Screenshot 2024-11-02 at 6 27 26 PM" src="https://github.com/user-attachments/assets/405c8b23-7d46-44bf-89c2-d733da583132">

### View Two Months at Once:
<img width="723" alt="Screenshot 2024-11-02 at 6 28 31 PM" src="https://github.com/user-attachments/assets/d9591939-4752-4d97-b22b-cae3898d65ef">

### Redesigned Presets:
<img width="1440" alt="Screenshot 2024-11-02 at 6 29 46 PM" src="https://github.com/user-attachments/assets/0406e1ec-5b9e-4bd6-82f0-cbd06f487585">

### Localization
| English | Arabic | Turkish |
|--------|-------|-------|
|<img width="551" alt="Screenshot 2024-10-26 at 9 09 20 PM" src="https://github.com/user-attachments/assets/1385d323-4326-43b2-a65b-5a510f17b6cc">|<img width="551" alt="Screenshot 2024-10-26 at 9 08 56 PM" src="https://github.com/user-attachments/assets/5fc4a5ac-77fe-4724-8e71-b0b99c2f92ec">|<img width="551" alt="Screenshot 2024-10-26 at 9 10 04 PM" src="https://github.com/user-attachments/assets/9816270b-f456-4766-8a1e-9c947bb652ff">|
> Note: There is no Range Selection in this pull request

## Types of Changes
- [ ] Chore (non-breaking change which improve code)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] Added story.
- [ ] Checked relative components.
- [ ] Checked the component on production.
- [ ] Added related docs.

AbdulrahmanNahhas reviewed (commented):
Here is a review of the changes in the pull request:

### General Overview
This pull request makes several updates to the handling and rendering of calendar components in the project. The changes include refactoring TypeScript components, updating import paths, adding new properties to components, and improving the localization and functionality of the calendar.

### Detailed Changes

1. **Component Update (`calendar.tsx`)**:
    - The `Calendar` component was converted from a function to an arrow function.
    - Changed import paths to use aliases for better modularization.
    - Updated the component's properties, improving its type safety and usage.

2. **Storybook Update (`advanced-calendar.stories.tsx`)**:
    - Added new controls (`argTypes`) to the Storybook configuration for the `AdvancedCalendar` component.
    - Updated the default `args` to include new properties such as `presets`, `view`, `fromYear`, `toYear`, and `type`.
    - Modified the `Template` to render with the new properties and to pass `args` to the component.

3. **Advanced Calendar Component Update (`index.tsx`)**:
    - Replaced usage of the `Select` component with a new `Popover` and `Command` based date selection.
    - Added new date options for quick selection.
    - Introduced new properties (`presets`, `view`, `fromYear`, `toYear`, `type`) to enhance the calendar's functionality.
    - Updated the component's state management and event handlers for better UX.

4. **Bug Fix in Filter Columns (`filter-column.tsx`)**:
    - Fixed an issue in the `GenerateFilterByType` function where the `toISOString` method was improperly used. The condition now checks if `toISOString` exists in the value before calling it.

### Positive Aspects
- The updates make the `Calendar` and `AdvancedCalendar` components more flexible and powerful.
- The changes improve code readability and maintainability by using aliases for imports and converting the component to an arrow function.
- Adding new controls in Storybook will make it easier for developers to test and understand the new features.
- The bug fix in the `GenerateFilterByType` function enhances robustness and prevents potential runtime errors.

### Areas for Improvement
- Ensure thorough testing of the new properties and functionalities in various scenarios to avoid regressions.
- Consider providing additional documentation or comments around the new properties to help other developers understand their usage and impact.

Overall, this pull request introduces valuable improvements to the calendar components, enhancing both functionality and developer experience.

a0m0rajab commented on src/molecules/tables/filter-column.tsx:
```diff
@@ -425,7 +425,9 @@ function GenerateFilterByType({
           mode="single"
           onSelect={(value) => {
-            setFilteredValue(value?.toISOString() || '');
+            if (value && 'toISOString' in value) {
+              setFilteredValue(value?.toString() || '');
+            }
```
I do not think we need this change here.

a0m0rajab commented on src/molecules/advanced-calendar/advanced-calendar.stories.tsx:
```diff
@@ -1,33 +1,52 @@
+    view: 'multiple',
   },
-  render: () => {
+  render: (args) => {
```
Nice that you added the args here 👍

a0m0rajab commented on src/molecules/advanced-calendar/index.tsx:
```diff
@@ -29,33 +39,83 @@ const getLocale = () => {
+
 export type AdvancedCalendarProps = CalendarProps & {
-  hideSelect?: boolean;
+  fromYear?: number;
```
since you have deleted the hideSelect, did you check if it's used anywhere else in the project?

AbdulrahmanNahhas replied:
Yes

a0m0rajab commented on src/molecules/advanced-calendar/index.tsx:
```diff
@@ -29,33 +39,83 @@ const getLocale = () => {
 export default function AdvancedCalendar({
-  hideSelect,
+  presets,
+  view,
```
can we set the default view as single?

a0m0rajab reviewed (changes requested):
Thanks for the PR, it looks good.  I left few comments, can you address them?

At the same time it would be great to change the PR title to follow conventional commits.

AbdulrahmanNahhas commented:
Thanks for the feedback! I’ve addressed the comments and made the requested changes. Let me know if there’s anything else you’d like me to adjust.

AbdulrahmanNahhas commented:
> Thanks for the PR, it looks good. I left few comments, can you address them?
> 
> At the same time it would be great to change the PR title to follow conventional commits.

Thanks for the feedback! I’ve addressed the comments and made the requested changes. Let me know if there’s anything else you’d like me to adjust.

a0m0rajab reviewed (approved):
LGTM Thanks for the PR. 

<img src="https://media1.giphy.com/media/tIeCLkB8geYtW/giphy.gif"/>
